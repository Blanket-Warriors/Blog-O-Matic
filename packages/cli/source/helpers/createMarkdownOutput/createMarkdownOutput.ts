import { load } from "js-yaml"
const Remarkable = require("remarkable")


export const remarkable = new Remarkable("commonmark", {
  breaks: true,
  html: true,
})


export async function createMarkdownOutput(
  text: string,
): Promise<[ { [key: string]: any }, string, string ]> {
  const parsed = /(?:^---\n)([\s\S]*)(?:---\n)(([\s\S])*)/gm.exec(text) || []
  const md = parsed[2]
    .replace(/\.jpeg/g, "-medium.jpeg")
    .replace(/\.jpg/g, "-medium.jpg")
    .replace(/\.png/g, "-medium.png")
    .replace(/\]\(\.\.\//g, "](../../")
  const frontmatter = load(parsed[1])
  const html = remarkable.render(md)

  return [ frontmatter, md, html ]
}
