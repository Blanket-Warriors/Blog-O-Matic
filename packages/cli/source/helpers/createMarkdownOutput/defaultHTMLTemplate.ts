export const template = `<html>
  <head>
    <% if (frontmatter) { %>
      <title><%= frontmatter.title %></title>
    <% } %>
  </head>
  <body>
    <%- blog %>
  </body>
</html>`
