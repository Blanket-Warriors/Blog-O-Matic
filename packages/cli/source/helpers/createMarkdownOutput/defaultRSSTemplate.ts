export const channelTemplate = `<?xml version="1.0" encoding="utf-8"?>
<rss version="2.0">
  <channel>
    <title><%= meta.title %></title>
    <link><%= meta.permalink %></link>
    <description><%= meta.description %></description>
    <% posts.forEach(function(post){ %>
      <item>
        <title><%= post.frontmatter.title %></title>
        <link><%= post.frontmatter.permalink %></link>
        <guid><%= post.frontmatter.permalink %></guid>
        <pubDate><%= post.frontmatter.releaseDate %></pubDate>
        <description><%- post.htmlBody %></description>
        </item>
      <item>
    <% }); %>
  </channel>
</rss>
`
