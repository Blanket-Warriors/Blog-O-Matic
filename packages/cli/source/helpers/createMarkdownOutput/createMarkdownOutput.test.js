import { advanceTo } from "jest-date-mock"
import { createMarkdownOutput } from "./createMarkdownOutput"
import mockPost from "./__mocks__/mockPost"


test("Should create markdown output", async () => {
  advanceTo(new Date(2018, 5, 27, 0, 0, 0))

  const [ frontmatter, md, html ] = await createMarkdownOutput(mockPost)

  expect(frontmatter).toMatchSnapshot()
  expect(md).toMatchSnapshot()
  expect(html).toMatchSnapshot()
})
