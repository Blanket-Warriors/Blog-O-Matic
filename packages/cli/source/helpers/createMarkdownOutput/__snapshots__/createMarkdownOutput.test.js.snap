// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should create markdown output 1`] = `
Object {
  "author": "ben",
  "created": 2018-12-22T00:59:57.015Z,
  "permalink": "post-2",
  "title": "Post 2",
  "updated": 2018-12-22T00:59:57.015Z,
}
`;

exports[`Should create markdown output 2`] = `
"# Post 2

## Ipse iamque

**Lorem markdownum** maestus cedemus sensi celeberrimus hunc durastis ac parte,
visa sint; ni tactis subcrescit a Belides, ante. Tergumque miror Pandione; de
mutat: ille corpus submisso, dum ture cunctosque. Tactis videri. **Pomo**
placet, cum quam sequuntur adamanteis duorum fueramque.

## Vestemque atris

Aditumque caeruleum urgues **manes** signa, me ad enim. Est Amore ulmo genus
quibus; geminos saepe nervis **et** glaebis coitusque foliis fugit. Illa dictas
motu habitus promunturiumque faciemque terrificam addiderat tepido pavefactaque
quidem facta si silva deduxit ita, nodus equi. Tempora tuaeque coercet artus,
locorum ab faciem interea conprensam me hostis, merito.

## Curat cessastis velox intercepta tenebrae

Et sparsit fero clavam honore indicium poterunt verba: ferire si leves. [Mare
fine vidit](http://dum-puta.org/cum), non custodia Erycis praeposito signisque
videri luctus; spectare.

![image](../../resources/image.large.png)

Miseri amor parabat Farfarus et diversa [quoque tulit](http://verbere.com/)
expugnacior, totidem. Sed colla tantum, eripies tua! Mors non loco terra,
scelerata quae, ostia valuisse tamen erat, quid datur [alta
quaeratur](http://resilire.org/haustos.html). Pectore et umbrarum non rapida.

## Quoque expalluit abstulit haec et nona superbia

**Et numen versus**, patefecit leto; timidae et *Delosque puppe ut* gente.
Myrtea et incompta Troum. Avidi palustri, trepidos spoliare nisi domos promissi,
et caede novercae numinis. Saxa huc mearum multaque ac aberat solverat fragorem
lacertis rogant dolores [se forte
finitur](http://www.primus.org/medio-defendite) his cervice nutrit.

    var megapixel_volume = name_windows(dvd_wiki_process(simplexCell),
            input_computer + diskDesktopText + 3);
    enterprise_enterprise_development(guid_shareware(bar + 1), -4 +
            art_network_warm + rpc_wheel_surface, 38 + 60 + 3);
    system = 213549;

## Sunt avem Pleuron Thaumantea vulgus morbo

Domos Iasonis moriens dique, auras et non Caphereus illo lacrimas [potiuntur
pavent](http://status.io/) amplexusque umero sequitur in possint. Atlantiades
siderea suis palla purpureas iam vetustas voluit?

    if (romDropKernel > captchaPanel) {
        portal_unicode_header += bittorrent_python_primary + wimaxMoodleIp(1,
                -3, 1);
    }
    dosRaw(telecommunications_association, pciHacker(
            terminal.editorMonitorFile.graphic(4, 2, access_remote)));
    bmp.mashup = dpi_batch_guid;

Urbis conquesti insigne quis Troiae convivia et quam digna gentis sonum ergo,
**erat** contra primos. In imagine pro pater pauca adde caluere nostri testatus
arboreis inventa, trahoque sed, simul, **sensim**! Materna semel, vox parens
illic, ipse senex, nimiumque nocte.
"
`;

exports[`Should create markdown output 3`] = `
"<html>
  <head>
    
      <title>Post 2</title>
    
  </head>
  <body>
    <h1>Post 2</h1>
<h2>Ipse iamque</h2>
<p><strong>Lorem markdownum</strong> maestus cedemus sensi celeberrimus hunc durastis ac parte,<br />
visa sint; ni tactis subcrescit a Belides, ante. Tergumque miror Pandione; de<br />
mutat: ille corpus submisso, dum ture cunctosque. Tactis videri. <strong>Pomo</strong><br />
placet, cum quam sequuntur adamanteis duorum fueramque.</p>
<h2>Vestemque atris</h2>
<p>Aditumque caeruleum urgues <strong>manes</strong> signa, me ad enim. Est Amore ulmo genus<br />
quibus; geminos saepe nervis <strong>et</strong> glaebis coitusque foliis fugit. Illa dictas<br />
motu habitus promunturiumque faciemque terrificam addiderat tepido pavefactaque<br />
quidem facta si silva deduxit ita, nodus equi. Tempora tuaeque coercet artus,<br />
locorum ab faciem interea conprensam me hostis, merito.</p>
<h2>Curat cessastis velox intercepta tenebrae</h2>
<p>Et sparsit fero clavam honore indicium poterunt verba: ferire si leves. <a href=\\"http://dum-puta.org/cum\\">Mare<br />
fine vidit</a>, non custodia Erycis praeposito signisque<br />
videri luctus; spectare.</p>
<p><img src=\\"../../resources/image.large.png\\" alt=\\"image\\" /></p>
<p>Miseri amor parabat Farfarus et diversa <a href=\\"http://verbere.com/\\">quoque tulit</a><br />
expugnacior, totidem. Sed colla tantum, eripies tua! Mors non loco terra,<br />
scelerata quae, ostia valuisse tamen erat, quid datur <a href=\\"http://resilire.org/haustos.html\\">alta<br />
quaeratur</a>. Pectore et umbrarum non rapida.</p>
<h2>Quoque expalluit abstulit haec et nona superbia</h2>
<p><strong>Et numen versus</strong>, patefecit leto; timidae et <em>Delosque puppe ut</em> gente.<br />
Myrtea et incompta Troum. Avidi palustri, trepidos spoliare nisi domos promissi,<br />
et caede novercae numinis. Saxa huc mearum multaque ac aberat solverat fragorem<br />
lacertis rogant dolores <a href=\\"http://www.primus.org/medio-defendite\\">se forte<br />
finitur</a> his cervice nutrit.</p>
<pre><code>var megapixel_volume = name_windows(dvd_wiki_process(simplexCell),
        input_computer + diskDesktopText + 3);
enterprise_enterprise_development(guid_shareware(bar + 1), -4 +
        art_network_warm + rpc_wheel_surface, 38 + 60 + 3);
system = 213549;
</code></pre>
<h2>Sunt avem Pleuron Thaumantea vulgus morbo</h2>
<p>Domos Iasonis moriens dique, auras et non Caphereus illo lacrimas <a href=\\"http://status.io/\\">potiuntur<br />
pavent</a> amplexusque umero sequitur in possint. Atlantiades<br />
siderea suis palla purpureas iam vetustas voluit?</p>
<pre><code>if (romDropKernel &gt; captchaPanel) {
    portal_unicode_header += bittorrent_python_primary + wimaxMoodleIp(1,
            -3, 1);
}
dosRaw(telecommunications_association, pciHacker(
        terminal.editorMonitorFile.graphic(4, 2, access_remote)));
bmp.mashup = dpi_batch_guid;
</code></pre>
<p>Urbis conquesti insigne quis Troiae convivia et quam digna gentis sonum ergo,<br />
<strong>erat</strong> contra primos. In imagine pro pater pauca adde caluere nostri testatus<br />
arboreis inventa, trahoque sed, simul, <strong>sensim</strong>! Materna semel, vox parens<br />
illic, ipse senex, nimiumque nocte.</p>

  </body>
</html>"
`;
